internal/registry/acctest/par.go

```diff
@@ -0,0 +1,328 @@
+	if err == nil {
+		return
+	}
+	if err, ok := err.(*packer_service.CreateBucketDefault); ok {
+		switch err.Code() {
+		case int(codes.AlreadyExists), http.StatusConflict:
+			// all good here !
+			return
+		}
+	}
+
+	cfg.T.Errorf("unexpected CreateBucket error, expected nil or 409. Got %v", err)
+}
+
+// UpsertIteration creates a new iteration if it does not already exists.
+func (cfg *Config) GetIterationByID(
+	bucketSlug,
+	id string,
+) string {
+	cfg.T.Helper()
+
+	getItParams := packer_service.NewGetIterationParams()
+	getItParams.LocationOrganizationID = cfg.Loc.OrganizationID
+	getItParams.LocationProjectID = cfg.Loc.ProjectID
+	getItParams.BucketSlug = bucketSlug
+	getItParams.IterationID = &id
+
+	ok, err := cfg.Packer.GetIteration(getItParams, nil)
+	if err != nil {
+		cfg.T.Fatal(err)
+	}
+	return ok.Payload.Iteration.ID
+}
+
+// UpsertIteration creates a new iteration if it does not already exists.
+func (cfg *Config) UpsertIteration(
+	bucketSlug,
+	fingerprint string,
+) string {
+	cfg.T.Helper()
+
+	_, err := cfg.CreateIteration(context.Background(), bucketSlug, fingerprint)
+	if err == nil {
+		return cfg.GetIterationIDFromFingerPrint(bucketSlug, fingerprint)
+	}
+	if err, ok := err.(*packer_service.CreateIterationDefault); ok {
+		switch err.Code() {
+		case int(codes.AlreadyExists), http.StatusConflict:
+			// all good here !
+			return cfg.GetIterationIDFromFingerPrint(bucketSlug, fingerprint)
+
+		}
+	}
+
+	cfg.T.Fatalf("unexpected CreateIteration error, expected nil or 409. Got %v", err)
+	return ""
+}
+
+// UpsertIteration creates a new iteration if it does not already exists.
+func (cfg *Config) MarkIterationAsDone(
+	bucketSlug,
+	iterID string,
+) {
+	cfg.T.Helper()
+
+	updateItParams := packer_service.NewUpdateIterationParams()
+	updateItParams.Body = &models.HashicorpCloudPackerUpdateIterationRequest{
+		BucketSlug:  bucketSlug,
+		IterationID: iterID,
+		Complete:    true,
+	}
+	updateItParams.IterationID = iterID
+	updateItParams.LocationOrganizationID = cfg.OrganizationID
+	updateItParams.LocationProjectID = cfg.ProjectID
+
+	_, err := cfg.Packer.UpdateIteration(updateItParams, nil)
+	if err == nil {
+		return
+	}
+
+	cfg.T.Errorf("unexpected UpdateIteration error: %v", err)
+}
+
+// GetIterationIDFromFingerPrint returns an iteration ID given its unique
+// fingerprincfg.t.
+func (cfg *Config) GetIterationIDFromFingerPrint(
+	bucketSlug,
+	fingerprint string,
+) string {
+	cfg.T.Helper()
+
+	getItParams := packer_service.NewGetIterationParams()
+	getItParams.LocationOrganizationID = cfg.Loc.OrganizationID
+	getItParams.LocationProjectID = cfg.Loc.ProjectID
+	getItParams.BucketSlug = bucketSlug
+	getItParams.Fingerprint = &fingerprint
+
+	ok, err := cfg.Packer.GetIteration(getItParams, nil)
+	if err != nil {
+		cfg.T.Fatal(err)
+	}
+	return ok.Payload.Iteration.ID
+}
+
+// UpsertBuild creates a new build for iteration if it does not already exists.
+func (cfg *Config) UpsertBuild(
```

Comment:
At first glance, It wasn't clear to me that this was the function function signature, especially for methods with multiple parameters of the same type. If you're aiming for readability, what do you think about passing in a structure?

Reply:
Good question here. And remember this is a test pkg and for readability of tests, I in other method calls: I tried to have the list be only a few arguments. Maybe this function is just going a little too much.